tests/e2e/tutorials/tutorialBase.js

```diff
@@ -208,6 +208,8 @@ class TutorialBase {
       await utils.sleep(2500);
       if (nodeIds.length === 0) {
         console.log("Services ready in", ((new Date().getTime())-start)/1000);
+        // after the service is responsive we need to a bit until the iframe is rendered
+        await utils.sleep(3000);
```

Comment:
We all know that these heuristics are very simple to solve immediate problems but very difficult to maintain and even justify (sometimes a sleep is a way to "hide" a real problem behind a green), i.e. why 3secs and before 2.5?? 

PROPOSAL:

Can we parametrize these sleeps to pre-define constants so that we can tune everything from a higher abstraction level? e.g. assume that we want our UI to be very responsive. We then classify and define WISHED specifications for each:
 - A response to an server request should be bounded between T_RESPONSE_MIN and T_RESPONSE_MAX
 - A service rendering in an iframe must render between T_IFRAME_RENDER_MIN and T_IFRAME_RENDER_MAX
...

Now, if all these sleeps in the tests are parametrized, we can at least reason on these heuristics such that a success in a test result guarantees e.g. that our services render iframes between this and that time or that the response of the servers are bound by other values.



In addition, instead of a simple sleep, we could use a retry between T_MIN and T_MAX.

That said, I created #1924.